experimental/instrumentation-fetch/src/instrumentation.ts

```diff
@@ -0,0 +1,84 @@
+import { BaseInstrumentation, faro, globalObject } from '@grafana/faro-core';
+
+import { fetchGlobalObjectKey, originalFetch, originalFetchGlobalObjectKey } from './constants';
+
+export class FetchInstrumentation extends BaseInstrumentation {
+  readonly name = '@grafana/faro-web-sdk:instrumentation-fetch';
+  readonly version = '1.0.0'; // TODO - pull from package.json
+
+  initialize(): void {
+    console.log('Initializing FetchInstrumentation');
+    this.enable();
+  }
+
+  constructor() {
+    super();
+  }
+
+  /**
+   * Instrument fetch with Faro
+   */
+  private _instrumentFetch(): typeof fetch {
```

Comment:
Q: `typeof fetch` will return `function`. 
Not sure if this is what you want or if you want the return type of `fetch()` as constructed by using teh TS utility type   `ReturnType<Type>`.

Reply:
I opted to return an entirely new function and overwrite the `fetch` property on the global object with this new function. Do you see any issues that might cause? I felt like it was a clean way to do it, but open to other options.